apps/web/pages/api/book/recurring-event.ts

```diff
@@ -53,16 +56,34 @@ async function handler(req: NextApiRequest & { userId?: number }, res: NextApiRe
       ...booking,
       appsStatus,
       allRecurringDates,
+      thirdPartyRecurringEventId,
+      numSlotsToCheckForAvailability,
       currentRecurringIndex: key,
       noEmail: key !== 0,
     };
 
-    const eachRecurringBooking = await handleNewBooking(recurringEventReq, {
-      isNotAnApiCall: true,
-      skipAvailabilityCheck: key >= numSlotsToCheckForAvailability,
-    });
+    const promiseEachRecurringBooking: ReturnType<typeof handleNewBooking> = handleNewBooking(
+      recurringEventReq,
+      {
+        isNotAnApiCall: true,
+        skipAvailabilityCheck: key >= numSlotsToCheckForAvailability || thirdPartyRecurringEventId !== null,
+      }
+    );
+
+    const eachRecurringBooking = await promiseEachRecurringBooking;
 
     createdBookings.push(eachRecurringBooking);
+
+    if (!thirdPartyRecurringEventId) {
+      if (eachRecurringBooking.references && eachRecurringBooking.references.length > 0) {
+        for (const reference of eachRecurringBooking.references!) {
+          if (reference.thirdPartyRecurringEventId) {
+            thirdPartyRecurringEventId = reference.thirdPartyRecurringEventId;
```

Comment:
It's more concise and readable
`thirdPartyRecurringEventId = eachRecurringBooking.references.find((reference)=>reference.thirdPartyRecurringEventId)?.thirdPartyRecurringEventId`